internal/internal_event_handlers.go

```diff
@@ -1072,9 +1075,17 @@ func (weh *workflowExecutionEventHandlerImpl) ProcessQuery(
 ) (*commonpb.Payloads, error) {
 	switch queryType {
 	case QueryTypeStackTrace:
-		return weh.encodeArg(weh.StackTrace())
+		if weh.enableEncodeBuiltInQueries {
```

Comment:
So I just asked the UI team how they know when to pass a payload to the remote codec. They said they pass all non-search-attributes to remote codecs. We ask people to pass through unencoded payloads in their codecs, but we don't necessarily require it.

Is there a concern that people whose remote codecs assume always-encoded and have worked in the UI today will now start breaking on stack traces? cc @bergundy, @Sushisource, and @Spikhalskiy. Just want confirmation that we can tell these users they need to fix their codec to pass through unencoded data. And if they come back with "but for security reasons we want to require encoding or fail" we are ok saying that they now need to restart their workers and opt-in with this boolean.

Reply:
If they already have remote codecs setup then the easier fix is to enable this option no? Assuming they read release notes (big assumption to be fair) they shouldn't need to restart their workers.

Reply:
> If they already have remote codecs setup then the easier fix is to enable this option no?

I agree, but it may not be obvious they need to enable this. So they'll go to the UI, see stack traces broken, and have to come back and redeploy all of their workers with this flag. I don't think it's that big of a deal, just want to get team confirmation here.

Reply:
I thought this was known that the codec should pass through anything it doesn't handle.
Could be worth documenting.

Reply:
People intentionally don't want to do that in all cases. Some users want to intentionally fail if they see unencoded data as a way to catch unencoded data early. So they'll need to set this field. We need to be very loud about this field because there are users that are very concerned about encrypted data automatically becoming unencrypted (however unfounded). We have even been asked for manual/security reviews that all non-search payloads are properly encrypted.

Reply:
I think the ideal situation would be able to set an option for the UI server like:

`payload_decoding: PassNonstandardToRemote | AlwaysPassToRemote` 

With the first being the default. This way users in the situation Chad talks about can choose to pass everything if they're paranoid (and can choose to do nothing if needed), but normally all known-decodable Payloads get handled locally.

Ofc, that's outside the scope of this change, but we could decide if we want to add that first to UI or not. Seems like it'd be an easy change.

Reply:
Another option would be to add a metadata field to this payload to mark that it is intentionally not encoded or have the UI provide more context to the remote codec server.

We need to make the proxy configurable to bypass stack trace queries which complicates this feature (granted we'll need to have it opt out for SDK generated metadata on WFT completion but the condition there is simpler).
Let's have a team discussion and see if we want to revert our decision to have this opt-in for encryption behavior.
Seems like we're adding a bunch of special cases where it might not be necessary.